### ISSUE TITLE
ModelPolisher cannot find any relations in ADB.

### ISSUE THREAD

**Issue by codekaust** [draeger-lab/ModelPolisher#56]
Current code in `feat/annotatedb` is not able to find relations in `adb`.
You can recreate error by building lightJar and running these commands:
```
docker build -t bigg:latest ./docker/bigg_docker/
```
```
docker build -t adb:latest ./docker/adb_docker/
```
```
docker run --name adb -p 1013:5432 -d adb:latest && docker run --name bigg -p 1310:5432 -d bigg:latest
```
```
java -jar ./target/ModelPolisher-noDB-1.7.jar --input=<input> --output=<output> --annotate-with-bigg=true --bigg-host=0.0.0.0 --bigg-port=1310 --add-adb-annotations=true --adb-host=0.0.0.0 --adb-port=1013
```

**Comment by matthiaskoenig** [comment]
@codekaust I will have a look at this in the next few hours

**Comment by matthiaskoenig** [comment]
This is now fixed. There were problems in your dockerfile and the postgres init script.
Mainly you did not create the correct users and so on so that the database dump was not put in the container. You should always look at the logs of your containers if running in detached mode, i.e., 
```
docker container logs adb
```
Shows you what went wrong.
Also always run `pg_restore` with the `-v` or `--verbose` flag to make sure things work.

After creating the container your should test if you can connect to the database. I did this with
idea.sh data sources (see below)

![pycharm_adb_connect](https://user-images.githubusercontent.com/900538/61953352-73031000-afb6-11e9-85b6-bfb2662664d1.png)

You can see the data in the docker container image
![pycharm_adb_content](https://user-images.githubusercontent.com/900538/61953397-8ca45780-afb6-11e9-8642-6e3e8e4cedaa.png)

Also don't map the ports in the docker files! only expose what must be exposed, i.e., 5432 in case of postgres

In summary, your scripts for restoring the db did not work, this is fixed now.

**Comment by codekaust** [comment]
Thank You @matthiaskoenig!
I will take care of your suggestions.

**Comment by codekaust** [comment]
Above is not the solution for this issue.
REASON: ModelPolisher is querying biggdb.

**Comment by matthiaskoenig** [comment]
Could you explain what the issue is? The relations are in the database, you just have to query them. If the issue is that ModelPolisher is not using the annotatedb relations then you should open a respective issue. But the relations are there and can be queried which fixes your issue description.

**Comment by codekaust** [comment]
Yes, the problem is not in ADB or in setting up of ADB by docker. The issue is somewhere in the connection of ModelPolisher with ADB container. I am trying to understand and resolve this issue, will update if I  find anything.

**Comment by codekaust** [comment]
Issue found: [Here](https://github.com/draeger-lab/ModelPolisher/blob/d2f79011182861ee1999992c472a8d5857eae129/src/main/java/edu/ucsd/sbrg/bigg/ModelPolisher.java#L424) we create connector for BGGDB and [here](https://github.com/draeger-lab/ModelPolisher/blob/d2f79011182861ee1999992c472a8d5857eae129/src/main/java/edu/ucsd/sbrg/bigg/ModelPolisher.java#L455) for ADB.

`connector.connect()` runs in each constructor but after one connector is connected(which happens for BiGGDB) the connect function for other connector is not run.

**Comment by codekaust** [comment]
```
PostgreSQLConnector psqlConnect = new PostgreSQLConnector(adb_host, new Integer(adb_port), adb_user, adb_passwd != null ? adb_passwd : "", adb_dbName);
          logger.info(psqlConnect.isConnected()+"");
          adb = new AnnotateDB(psqlConnect);
```
2nd line gives result true. See [here](https://github.com/draeger-lab/ModelPolisher/blob/d2f79011182861ee1999992c472a8d5857eae129/src/main/java/edu/ucsd/sbrg/bigg/ModelPolisher.java#L455).

**Comment by draeger** [comment]
It seems to me this is a problem of ADB.

**Comment by codekaust** [comment]
No, it was a problem with SQLConnector only as I have mentioned in the above comments.
